File: odemapster-r2rml-rdb/src/main/java/es/upm/fi/dia/oeg/obdi/wrapper/r2rml/rdb/model/R2RMLObjectMap.java
package es.upm.fi.dia.oeg.obdi.wrapper.r2rml.rdb.model;

import com.hp.hpl.jena.rdf.model.Resource;

import es.upm.fi.dia.oeg.obdi.wrapper.r2rml.rdb.exception.R2RMLInvalidTermMapException;
import es.upm.fi.dia.oeg.obdi.wrapper.r2rml.rdb.model.R2RMLTermMap.TermMapType;

public class R2RMLObjectMap extends R2RMLTermMap {
	
	public static R2RMLObjectMap create(Resource resource, R2RMLTriplesMap owner) 
			throws R2RMLInvalidTermMapException {
		R2RMLObjectMap om = new R2RMLObjectMap();
		om.parse(resource, owner);
		return om;
	}
	
//	public R2RMLObjectMap(String constantValue) {
//		super(TermMapPosition.OBJECT, constantValue);		
//	}
	
	public static R2RMLObjectMap create(String constantValue) {
		R2RMLObjectMap om = new R2RMLObjectMap();
		om.termMapType = TermMapType.CONSTANT;
		om.constantValue = constantValue;
		return om;
	}	
}

File: morph-r2rml/src/main/scala/es/upm/dia/fi/oeg/morph/r2rml/model/R2RMLObjectMap.scala
package es.upm.dia.fi.oeg.morph.r2rml.model

import scala.collection.JavaConversions._
import org.apache.log4j.Logger
import com.hp.hpl.jena.rdf.model.Resource
import es.upm.fi.dia.oeg.morph.base.Constants
import com.hp.hpl.jena.rdf.model.RDFNode

class R2RMLObjectMap(termMapType:Constants.MorphTermMapType.Value
    , termType:Option[String], datatype:Option[String], languageTag:Option[String]) 
    extends R2RMLTermMap(termMapType, termType, datatype, languageTag) {
  
	override val logger = Logger.getLogger(this.getClass().getName());

}

object R2RMLObjectMap {
	def apply(rdfNode:RDFNode) : R2RMLObjectMap = {
		val coreProperties = R2RMLTermMap.extractCoreProperties(rdfNode);
		//coreProperties = (termMapType, termType, datatype, languageTag)
		val termMapType = coreProperties._1;
		val termType = coreProperties._2;
		val datatype = coreProperties._3;
		val languageTag = coreProperties._4;
		val om = new R2RMLObjectMap(termMapType, termType, datatype, languageTag);
		om.parse(rdfNode);
		om;	  
	}
	
	def extractObjectMaps(resource:Resource) : Set[R2RMLObjectMap] = {
	  val tms = R2RMLTermMap.extractTermMaps(resource, Constants.MorphPOS.obj);
	  val result = tms.map(tm => tm.asInstanceOf[R2RMLObjectMap]);
	  result;
	}
}